mesonbuild/wrap/wrap.py

```diff
@@ -226,16 +226,21 @@ def get_file(self):
             self.apply_patch()
 
     def get_git(self):
+        # git doesn't support directly cloning shallowly for commits,
+        # so we follow https://stackoverflow.com/a/43136160
+        subprocess.check_call(['git', 'init', self.directory], cwd=self.subdir_root)
+        subprocess.check_call(['git', 'remote', 'add', 'origin', self.wrap.get('url')],
+                               cwd=self.dirname)
+        git_options = []
         revno = self.wrap.get('revision')
-        subprocess.check_call(['git', 'clone', self.wrap.get('url'),
-                               self.directory], cwd=self.subdir_root)
-        if revno.lower() != 'head':
-            if subprocess.call(['git', 'checkout', revno], cwd=self.dirname) != 0:
-                subprocess.check_call(['git', 'fetch', self.wrap.get('url'), revno], cwd=self.dirname)
-                subprocess.check_call(['git', 'checkout', revno], cwd=self.dirname)
+        if self.wrap.values.get('depth', '') != '':
+            git_options.extend(['--depth', self.wrap.values.get('depth')])
+        subprocess.check_call(['git', 'fetch'] + git_options + ['origin', revno],
+                cwd=self.dirname)
```

Comment:
__[Flake8]__ [E128] continuation line under-indented for visual indent ([link](https://sider.review/gh/repos/19784232/pulls/5932#issue-4580791))

<sub>posted by Sider</sub>

Reply:
How do I know what the right level of indentation would be?